platform_plugin_turnitin/turnitin.py

```diff
@@ -0,0 +1,426 @@
+                    "id": user_data["user_id"],
+                    "given_name": user_name[0] if user_name else "no_name",
+                    "family_name": " ".join(user_name[1:])
+                    if len(user_name) > 1
+                    else "no_last_name",
+                    "email": user_data["user_email"],
+                },
+                "original_submitted_time": date_now,
+            },
+        }
+        return post_create_submission(payload)
+
+    @XBlock.handler
+    def upload_turnitin_submission_file(self, data, suffix=""):
+        """
+        Handles the upload of the user's file to Turnitin.
+
+        Args:
+            data (WebRequest): Web request containing the file to be uploaded.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            Response: The response after uploading the file to Turnitin.
+        """
+        turnitin_submission = self.create_turnitin_submission_object()
+        if turnitin_submission.status_code == 201:
+            turnitin_submission_id = turnitin_submission.json()["id"]
+            current_user_id = self.get_user_data()["user_id"]
+            current_user = User.objects.get(id=current_user_id)
+            submission = TurnitinSubmission(
+                user=current_user, turnitin_submission_id=turnitin_submission_id
+            )
+            submission.save()
+            myfile = data.params["myfile"].file
+            # turnitin_submission_id='0a966646-83f9-4ce6-aa47-71e07baf4e30'
+            response = put_upload_submission_file_content(
+                turnitin_submission_id, myfile
+            )
+            return Response(
+                json.dumps(response.json()),
+                content_type="application/json",
+                charset="UTF-8",
+            )
+        return Response(
+            json.dumps(turnitin_submission.json()),
+            content_type="application/json",
+            charset="UTF-8",
+        )
+
+    @XBlock.json_handler
+    def get_submission_status(self, data, suffix=""):
+        """
+        Retrieves the status of the latest Turnitin submission for the user.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: Information related to the user's latest Turnitin submission.
+        """
+        current_user_id = self.get_user_data()["user_id"]
+        current_user = User.objects.get(id=current_user_id)
+        try:
+            last_submission = TurnitinSubmission.objects.filter(
+                user=current_user
+            ).latest("created_at")
+        except TurnitinSubmission.DoesNotExist:
+            return {"success": False}
+        # last_submission = 'de6784c5-471f-4220-aff1-16b6b44dffcf'
+        response = get_submission_info(last_submission.turnitin_submission_id)
+        return response.json()
```

Comment:
Can we explicitly return a dictionary? so we know what we're returning. Right now I'm not sure what response.json() returns